test/tls.c

```diff
@@ -61,7 +61,7 @@ volatile int32_t *volatile overaligned_data_addr;
 volatile int32_t *volatile bss_addr;
 volatile int32_t *volatile overaligned_bss_addr;
 
-#ifdef __THREAD_LOCAL_STORAGE_API
+#if defined(__SEMIHOST) && defined(__THREAD_LOCAL_STORAGE_API)
```

Comment:
What's the check for __SEMIHOST for?

Reply:
__THREAD_LOCAL_STORAGE_API shouldn't be set for native configurations though, I think I just changed the check in this code from __THREAD_LOCAL_STORAGE to __THREAD_LOCAL_STORAGE_API.

Reply:
Oof. Yeah, we need to make sure it isn't set in the native case -- we can't use that code as the native C library will be in charge of the TLS block. Suggestions welcome as to how to make the configuration work in this case.

Reply:
the TLS API doesn't depend upon semihost though, it should only depend upon have_picolibc_tls_api as that's all we actually care about. I think the issue you're hitting is that when you provide x86 TLS api code, that gets found when building native targets by the simplistic check we're using. That API is only actually available if picolib is built, and that doesn't happen on native targets. So, have_picolibc_tls_api needs to be set to false when picolib isn't available.